### ISSUE TITLE
Document json_types module

### ISSUE THREAD

**Issue by mikedotexe** [near/near-sdk-rs#359]
View the current Rust docs here:
https://docs.rs/near-sdk/3.1.0/near_sdk/json_types/index.html

- [ ] Expand description (Currently has "Helper types for JSON serialization.") Explain why JSON serialization is needed. Some thoughts… AssemblyScript needs it and we want both languages to interoperate. How an off-chain call will use JSON serialization to transmit data on-chain. How return values are similar (on-chain » off-chain this time).
- [ ] Add descriptions for I64, I128, U64, U128 (Include an example of how this breaks, can take this from simple contract here https://github.com/mikedotexe/near-debug-rust-u128)
- [ ] Inside `ValidAccountId` (https://docs.rs/near-sdk/3.1.0/near_sdk/json_types/struct.ValidAccountId.html) expand description and common usages. If you're not familiar, globally search this repo and https://github.com/near/core-contracts for `ValidAccountId` and see how they're used. Reach out if there are questions.
- [ ] Inside `Base64VecU8` (https://docs.rs/near-sdk/3.1.0/near_sdk/json_types/struct.Base64VecU8.html) expand description and show usage examples. Again, globally search for it. A reference implementation that comes to mind: https://github.com/Cron-Near/contracts/blob/main/manager/src/lib.rs

**Comment by matklad** [comment]
Hm, having looked at the types in question, I doubt many of them make sense in the ideal world. For example, `type WrappedBalance = U128;` doesn't really make sense. 

What I'd expect to see in the sdk is

```
pub struct Ballance(pub u128);
```

Similarly, `ValidAccountId` doesn't make sense -- it should be impossible to represent invalid account id in the first place. 

I think these things exist because sdk directly use nearcore types, which come with a historical baggage and from the context of a binary project, where nice API and version compat does not matter. 

If I were a king of the world, I'd order to refactor these types away, rather than documenting current (mostly accidental) state. Seems less total work that way. 

But it might be the case that it make more sense to document current state as is, rather than spend much more time redesigning. 

I just want to raise the issue that, from the perspective of nice Rust APIs (https://rust-lang.github.io/api-guidelines/about.html), the current offering seems rather ungood :-)

**Comment by volovyk-s** [comment]
Good point @matklad. But probably it will be a breaking change. @mikedotexe @evgenykuzyakov need your opinion here.

**Comment by matklad** [comment]
@volovyk-s opened https://github.com/near/near-sdk-rs/issues/384 to discuss the underlying issue.

**Comment by mikedotexe** [comment]
As discussed with Serhii, we're going to hold off on doing work on this issue until next week when there's been a discussion with Austin, Kladov, and Eugene.
